src/components/database-replica.ts

```diff
@@ -0,0 +1,130 @@
+import * as aws from '@pulumi/aws';
+import * as pulumi from '@pulumi/pulumi';
+import { commonTags } from '../constants';
+
+export type DatabaseReplicaArgs = {
+  /**
+   * ARN of the primary DB that we want to replicate.
+   */
+  replicateSourceDb: pulumi.Input<string>;
+  /**
+   * DB subnet group name. Should be the same as primary instance.
+   * * If primary DB is instance of studion:Database, it can be accessed as
+   * `db.dbSubnetGroup.name`.
+   */
+  dbSubnetGroupName: pulumi.Input<string>;
+  /**
+   * DB security group ID. Should be the same as primary instance.
+   * If primary DB is instance of studion:Database, it can be accessed as
+   * `db.dbSecurityGroup.id`.
+   */
+  dbSecurityGroupId: pulumi.Input<string>;
+  /**
+   * IAM Monitoring role. Should be the same as primary instance.
+   */
+  monitoringRole?: aws.iam.Role;
+  /**
+   * Specifies if the RDS instance is multi-AZ. Defaults to false.
+   */
+  multiAz?: pulumi.Input<boolean>;
+  /**
+   * Specifies whether any database modifications are applied immediately,
+   * or during the next maintenance window. Default is false.
+   */
+  applyImmediately?: pulumi.Input<boolean>;
+  /**
+   * The allocated storage in gibibytes. Defaults to 20GB.
+   */
+  allocatedStorage?: pulumi.Input<number>;
+  /**
+   * The upper limit to which Amazon RDS can automatically scale
+   * the storage of the DB instance. Defaults to 100GB.
+   */
+  maxAllocatedStorage?: pulumi.Input<number>;
+  /**
+   * The instance type of the RDS instance. Defaults to 'db.t4g.micro'.
+   */
+  instanceClass?: pulumi.Input<string>;
+  /**
+   * Set this to true to enable database monitoring. Defaults to false.
+   */
+  enableMonitoring?: pulumi.Input<boolean>;
+  /**
+   * A map of tags to assign to the resource.
+   */
+  tags?: pulumi.Input<{
+    [key: string]: pulumi.Input<string>;
+  }>;
+};
+
+const defaults = {
```

Comment:
@ikovac should I put defaults from this and the Database component to constants file and reuse or do we want to have them separate? I just noticed we have `skipFinalSnapshot` here which does not apply to replica.

Also, based on our offline talk, replicas can have multiAz too.